test/test-trace-agent.js

```diff
@@ -34,17 +34,13 @@ var assert = require('assert');
 var config = require('../config.js');
 var file = require('../src/trace-agent.js');
 var SpanData = require('../src/span-data.js');
-// TODO: This line will silently fail if a logger is not supplied as
-//       the second argument.
```

Comment:
The second argument not being present is a legitimate failure condition for the agent. Before the TODO was introduced the tests did not pass a logger but it worked out due to the limited use of the agent in the tests. Adding the mock logger addresses this issue.